web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -66,18 +71,18 @@ class OfficeHoursCalendarView(ImpactView):
     view_name = "office_hours_calendar_view"
     SUCCESS_HEADER = "Office hours fetched successfully"
     FAIL_HEADER = "Office hours could not be fetched"
-    BAD_DATE_SPEC = "We were unable to parse the date specifier"
+    BAD_FOCAL_DATE = "We were unable to parse the date specifier"
     NO_SUCH_USER = "We were not able to locate that user"
     NOT_OFFICE_HOURS_VIEWER = ("You are not able to view office hours at this "
                                "time. Please see MassChallenge staff.")
 
     def get(self, request):
         self.response_elements = {}
         (self._get_target_user(request) and
-         self._check_request_user_type(request) and
+         self._check_request_user_type() and
          self._get_date_range(request) and
          self._set_user_query() and
-         self._get_office_hours_data())
+         self._get_office_hours_data(request))
         return Response(self.response_elements)
 
     def _get_target_user(self, request):
```

Comment:
We no longer need the employee check. This logics to me means that if the user making the request is not a staff user and the user_id that has been passed is not there own, return a permission error. We need finalists to be able to fetch a specific mentor's office hours too. cc @jonkiparsky

Reply:
@jonkiparsky what is your take on the above?

Reply:
Yes, that makes sense